its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/SslTest.java

```diff
@@ -219,25 +219,20 @@ void untrustedSelfSignedCertificate() throws Exception {
 
     Path projectDir = TestUtils.projectDir(basePath, "ProjectUnderTest");
 
-    ScannerForMSBuild build = TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
+    var result = TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
       .addArgument("begin")
       .setProjectKey(projectKey)
       .setProperty("sonar.host.url", server.getUrl())
       .setProjectName("sample")
       .setProperty("sonar.projectBaseDir", projectDir.toAbsolutePath().toString())
-      .setProjectVersion("1.0");
-
-    try {
-      ORCHESTRATOR.executeBuild(build);
-      fail("Expecting to fail during the begin with an SSL error");
-    } catch (BuildFailureException e) {
-      assertFalse(e.getResult().isSuccess());
-      assertThat(e.getResult().getLogs())
-        .contains("System.Net.WebException: The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.")
-        .contains("System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure.");
-    } finally {
-      server.stop();
-    }
+      .setProjectVersion("1.0")
+      .execute(ORCHESTRATOR);
+
+    assertFalse(result.isSuccess());
+    assertThat(result.getLogs())
+      .contains("System.Net.WebException: The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.")
+      .contains("System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure.");
+    server.stop();
```

Comment:
Didn't @sebastien-marichal had some PR with [auto-close](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) opened that made sure `server.stop()` is always executed? I actually liked that approach.